defs.h

```diff
@@ -133,10 +133,12 @@
 
 #define MemFreeDbg(p)	\
 {						\
-	void *p__p;			\
-	p__p = p;			\
-	p = NULL;			\
-	mem_free_dbg(p__p);	\
+	if (p) {\

```

Comment:
Is this change needed for some reason?

Reply:
I assume it was a crash related to p already being null. So perhaps an assert is needed?

Reply:
This helped me with byte swapping.
There are still  0 freeings even on Little Endian.

Reply:
mem_free_dbg it self checks if it's 0 so there isn't any need to also check it here.
https://github.com/diasurgical/devilutionX/blob/e88e574c5f1e537007bb634873ea90270d92d9fd/Source/engine.cpp#L1495

Reply:
Ah, remember now, , I had "else { printf() }" here to find free of null .